mgplvm/crossval/crossval.py

```diff
@@ -0,0 +1,130 @@
+import os
+import numpy as np
+import copy
+import mgplvm
+import torch
+from mgplvm import kernels, rdist, models, training
+from mgplvm.manifolds import Torus, Euclid, So3
+import matplotlib.pyplot as plt
+import pickle
+from scipy.stats import ttest_1samp
+from . import train_model, load_model
+torch.set_default_dtype(torch.float64)
+
+def not_in(arr, inds):
+    mask = np.ones(arr.size, dtype=bool)
+    mask[inds] = False
+    return arr[mask]
+
+def update_params(params, **kwargs):
+    newps = copy.copy(params)
+    for key, value in kwargs.items():
+        newps[key] = value
+    return newps
+
+def train_cv(Y,
+            device,
+            model_ps,
+            train_ps,
+            T1 = None,
+            N1 = None,
+            nt_train = None,
+            nn_train = None,
+            test = True):
+    """
+    Parameters
+    ----------
+    Y : array
+        data with dimensionality (n x m x n_samples)
+    device : torch.device
+        GPU/CPU device on which to run the calculations
+    model_ps : dict
```

Comment:
do we want to pass in model_ps? or do we want to pass in a model directly? not sure which one is better